### PR TITLE
builtins: fix ceil and floor builtins not working with integer input

### LINKED ISSUE #45999: sql: allow floor function to operate on ints
In Postgres, we see this (the input is int, and the output seems to be decimal:
```
postgres> select floor(2);
+---------+
| floor   |
|---------|
| 2.0     |
+---------+
```
In CockroachDB we see this:
```
> select floor(2);
ERROR: unknown signature: floor(int)
SQLSTATE: 42883
```

### PR DESCRIPTION
**Pull request by pohzipohzi**
Fixes #45999

Previously, these functions were returning an error:

```
> SELECT floor(1);
ERROR: unknown signature: floor(int)
SQLSTATE: 42883
```

With this change we are now able to do:

```
> SELECT floor(1);
  floor
---------
      1
(1 row)
```

Also standardises description for the two related builtins.

Release justification: low risk improvement to current functionality

Release note (sql change): This PR allows the ceil and floor builtins to
accept integer inputs.

### CODE REVIEW

File: pkg/sql/sem/builtins/builtins.go [hunk 1]
```diff
@@ -2160,6 +2160,14 @@ may increase either contention or retry errors, or both.`,
 			_, err := tree.ExactCtx.Floor(&dd.Decimal, x)
 			return dd, err
 		}, "Calculates the largest integer not greater than `val`."),
+		tree.Overload{
```
Comment (rohany) [inline pkg/sql/sem/builtins/builtins.go]: This should use a similar overload builder like the cases for float and decimal

File: pkg/sql/sem/builtins/builtins.go [hunk 2]
```diff
@@ -3853,15 +3861,23 @@ var uuidV4Impl = makeBuiltin(
...
 	decimalOverload1(func(x *apd.Decimal) (tree.Datum, error) {
 		dd := &tree.DDecimal{}
 		_, err := tree.ExactCtx.Ceil(&dd.Decimal, x)
 		if dd.IsZero() {
 			dd.Negative = false
 		}
 		return dd, err
-	}, "Calculates the smallest integer greater than `val`."),
+	}, "Calculates the smallest integer not smaller than `val`."),
+	tree.Overload{
```
Comment (rohany) [inline pkg/sql/sem/builtins/builtins.go]: Ditto as above

**Review by rohany — CHANGES_REQUESTED** [review]
Thanks for your contribution! A few nits:

* Can you update your release justification to be "low risk improvement to current functionality"
* Can you update your release note to be "This PR allows the `ceil` and `floor` builtins to accept integer inputs."
* Can you update the PR message to denote that this PR fixes the linked issue?

**Comment by pohzipohzi** [comment]
Thanks for the review - tried to address the comments :slightly_smiling_face: 

I see that there's a failing test but I believe it should be unrelated to this PR

**Comment by rohany** [comment]
Thanks for your contribution! This looks great. I'll rerun the CI and then merge this after.

**Comment by otan** [comment]
bors r-

sorry i just took a look at this - it's not quite correct :) details coming!

File: pkg/sql/sem/builtins/builtins.go [hunk 3]
```diff
@@ -2160,6 +2160,9 @@ may increase either contention or retry errors, or both.`,
 			_, err := tree.ExactCtx.Floor(&dd.Decimal, x)
 			return dd, err
 		}, "Calculates the largest integer not greater than `val`."),
+		integerOverload1(func(x int64) (tree.Datum, error) {
+			return tree.NewDInt(tree.DInt(x)), nil
+		}, "Calculates the largest integer not greater than `val`."),
```
Comment (otan) [inline pkg/sql/sem/builtins/builtins.go]: the return type of this function is float --
```
otan=# select pg_typeof(floor(2::int));
    pg_typeof
------------------
 double precision
(1 row)
```

also, do you mind making this work for floating point arithmetic?

postgres:
```
otan=# select pg_typeof(floor(2::float));
    pg_typeof
------------------
 double precision
(1 row)
```

with your change i think it still doesn't work

same with ceil!

**Review by otan — CHANGES_REQUESTED** [review]

**Review by otan — APPROVED** [review]
thanks!

**Comment by otan** [comment]
bors r=otan,rohany

File: pkg/sql/sem/builtins/builtins.go (hunk 3 above)
Comment (pohzipohzi) [reply pkg/sql/sem/builtins/builtins.go]: Thanks for the review, and sorry about the mistake. I've tried to make a fix for this.

Not sure if i'm understanding the part about floating point arithmetic correctly. Running `select pg_typeof(floor(2::float));` returns `float` with the current implementation. Could you provide a test case to show what you mean?

Comment (otan) [reply pkg/sql/sem/builtins/builtins.go]: > sorry about the mistake
don't be sorry :) very hard to spot!

> Not sure if i'm understanding the part about floating point arithmetic correctly. Running select pg_typeof(floor(2::float)); returns float with the current implementation. Could you provide a test case to show what you mean?

i noticed you added a test for it and it was already correct (i missed that part of the code), so disregard that comment.

thanks again for doing this.

**Comment by SWDevAngel** [comment]
@pohzipohzi Hi Poh Zi How. Thank you for contributing to CockroachDB this year. As a token of our appreciation we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) `@lisa` so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please send this asap. :)

**Pull request merged**